admin-metrics/src/main/kotlin/com/epam/drill/admin/metrics/config/MetricsScheduler.kt

```diff
@@ -0,0 +1,96 @@
+ */
+package com.epam.drill.admin.metrics.config
+
+import com.epam.drill.admin.metrics.job.RefreshMaterializedViewJob
+import com.epam.drill.admin.metrics.job.VIEW_NAME
+import org.kodein.di.DI
+import org.quartz.*
+import org.quartz.impl.StdSchedulerFactory
+import org.quartz.impl.jdbcjobstore.JobStoreTX
+import org.quartz.impl.jdbcjobstore.PostgreSQLDelegate
+import org.quartz.simpl.SimpleThreadPool
+import org.quartz.utils.ConnectionProvider
+import org.quartz.utils.DBConnectionManager
+import java.util.*
+import javax.sql.DataSource
+
+class MetricsScheduler(
+    private val config: SchedulerConfig
+) {
+    private lateinit var scheduler: Scheduler
+
+    fun init(di: DI, dataSource: DataSource) {
+        val quartzProperties = Properties().apply {
+            setProperty("org.quartz.scheduler.instanceName", "MetricsScheduler")
+            setProperty("org.quartz.scheduler.instanceId", "AUTO")
+            setProperty("org.quartz.threadPool.class", SimpleThreadPool::class.java.name)
+            setProperty("org.quartz.threadPool.threadCount", config.threadPools.toString())
+            setProperty("org.quartz.jobStore.class", JobStoreTX::class.java.name)
+            setProperty("org.quartz.jobStore.driverDelegateClass", PostgreSQLDelegate::class.java.name)
+            setProperty("org.quartz.jobStore.isClustered", "true")
+
+            setProperty("org.quartz.jobStore.tablePrefix", "quartz.")
+            setProperty("org.quartz.jobStore.dataSource", "schedulerDS")
+        }
+
+        val schedulerFactory = StdSchedulerFactory(quartzProperties)
+        val provider = QuartzDataSourceConnectionProvider(dataSource)
+        DBConnectionManager.getInstance().addConnectionProvider("schedulerDS", provider)
+
+        scheduler = schedulerFactory.scheduler
+        scheduler.setJobFactory(KodeinJobFactory(di))
+    }
+
+    fun start() {
+        val refreshMethodsCoverageViewJob = JobBuilder.newJob(RefreshMaterializedViewJob::class.java)
+            .withIdentity("refreshMethodsCoverageViewJob", "refreshMaterializedViews")
+            .usingJobData(VIEW_NAME, "matview_methods_coverage")
+            .build()
+
+        val refreshViewIntervalTrigger = TriggerBuilder.newTrigger()
+            .startNow()
+            .withSchedule(
+                SimpleScheduleBuilder.simpleSchedule()
+                    .withIntervalInMinutes(config.refreshViewsIntervalInMinutes)
+                    .repeatForever()
+                    .withMisfireHandlingInstructionIgnoreMisfires()
+            )
+            .build()
+
+        scheduler.start()
+
+        scheduleJob(refreshMethodsCoverageViewJob, refreshViewIntervalTrigger)
+    }
+
+    fun scheduleJob(jobDetail: JobDetail, trigger: Trigger) {
+        if (!scheduler.checkExists(jobDetail.key)){
+            scheduler.scheduleJob(jobDetail, trigger)
+        }
+    }
+
+    fun shutdown() {
+        if (!scheduler.isShutdown) {
+            scheduler.shutdown(false)
```

Comment:
When shutting down the server, we can wait for all jobs to complete or not. 
```
scheduler.shutdown(waitForJobsToComplete = false)
````
I chose not to wait, because if the job takes a long time to complete, then we will have to kill the process with the `kill -9` command.